x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformState.java

```diff
@@ -65,24 +81,37 @@
             throw new IllegalArgumentException("Unsupported token [" + p.currentToken() + "]");
         }, CURRENT_POSITION, ObjectParser.ValueType.VALUE_OBJECT_ARRAY);
         PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), GENERATION);
```

Comment:
This argument cannot be optional because the constructor argument is `long`, which cannot be `null`.  So either this is compulsory or the constructor argument must be `Long`.